feature-pack-api/src/main/java/org/jboss/provisioning/xml/ProvisionedConfigXmlUtil.java

```diff
@@ -335,7 +335,7 @@ private static void readParam(XMLExtendedStreamReader reader, final ProvisionedF
         } if(value == null) {
             throw ParsingUtils.missingAttributes(reader.getLocation(), Collections.singleton(Attribute.VALUE));
         }
-        featureBuilder.setConfigParam(name, value);
+        featureBuilder.setResolvedParam(name, value);
```

Comment:
Here I'm not sure - the writer is writing only resolvedParams: https://github.com/aloubyansky/pm/blob/master/feature-pack-api/src/main/java/org/jboss/provisioning/xml/ProvisionedConfigXmlWriter.java#L70

so shouldn't the params be read as resolved again instead of config?

Reply:
It should be the config value in both cases. Parameters may be of non-string types that require a format other than the toString() method produces. feature.getConfigValue(param) is supposed to return the properly string value. The following should work

                for(String param : feature.getParamNames()) {
                    final ElementNode paramE = addElement(featureE, Element.PARAM);
                    addAttribute(paramE, Attribute.NAME, param);
                    addAttribute(paramE, Attribute.VALUE, feature.getConfigParam(param));
                }

Thanks for spotting that!